drivers/modbus_rtu/include/modbus_rtu.h

```diff
@@ -0,0 +1,173 @@
+#pragma once
+
+/**
+ * @defgroup    drivers_modbus ModBus
+ * @ingroup     drivers
+ * @brief       Modbus
+ *
+ * @{
+ * @file
+ * @brief       Modbus interface definitions
+ *
+ * @author      Denis Litvinov <li.denis.iv@gmail.com>
+ */
+
+#include "periph/uart.h"
+#include "thread.h"
+#include "mutex.h"
+
+/**
+ * @brief Modbus functions
+ */
+enum MB_FC {
+  MB_FC_NONE = 0,                /**< null operator */
+  MB_FC_READ_COILS = 1,          /**< FCT=1 -> read coils or digital outputs */
+  MB_FC_READ_DISCRETE_INPUT = 2, /**< FCT=2 -> read digital inputs */
+  MB_FC_READ_REGISTERS = 3,      /**< FCT=3 -> read registers or analog outputs */
+  MB_FC_READ_INPUT_REGISTER = 4, /**< FCT=4 -> read analog inputs */
+  MB_FC_WRITE_COIL = 5,          /**< FCT=5 -> write single coil or output */
+  MB_FC_WRITE_REGISTER = 6,      /**< FCT=6 -> write single register */
+  MB_FC_WRITE_COILS = 15,        /**< FCT=15 -> write multiple coils or outputs */
+  MB_FC_WRITE_REGISTERS = 16     /**< FCT=16 -> write multiple registers */
+};
+
+//! Максимальный размер пакета протокола Modbus RTU.
+#define MODBUS_RTU_PACKET_SIZE_MAX 256
+
+//! Размер контрольной суммы протокола Modbus RTU.
+#define MODBUS_RTU_CRC_SIZE 2
+
+//! Размер полей сообщения протокола Modbus RTU.
+#define MODBUS_RTU_FIELDS_SIZE (sizeof(modbus_rtu_id_t) + sizeof(modbus_rtu_func_t))
+//! Размер полей и контрольной суммы сообщения протокола Modbus RTU.
+#define MODBUS_RTU_FIELDS_CRC_SIZE (MODBUS_RTU_CRC_SIZE + MODBUS_RTU_FIELDS_SIZE)
+//! Максимальный размер данных протокола Modbus RTU.
+#define MODBUS_RTU_DATA_SIZE_MAX (MODBUS_RTU_PACKET_SIZE_MAX - MODBUS_RTU_FIELDS_CRC_SIZE)
+//! Максимальный размер данных и контрольной суммы протокола Modbus RTU.
+#define MODBUS_RTU_DATA_CRC_SIZE_MAX (MODBUS_RTU_DATA_SIZE_MAX + MODBUS_RTU_CRC_SIZE)
+
+/**
+ * @brief Modbus message structure.
+ */
+typedef struct {
+  uint8_t id;     /**< ID of slave, on response not use */
+  uint8_t func;   /**< function code */
+  uint16_t addr;  /**< Starting Address */
+  uint16_t count; /**< Quantity of Registers or byte or bits */
+  uint16_t *regs; /**< Registers or byte or bits*/
+} modbus_rtu_message_t;
+
+/**
+ * @brief Modbus structure.
+ */
+typedef struct {
+  /**
+   * @brief  UART port.
+   */
+  uart_t uart;
+  /**
+   * @brief  baudrate.
+   */
+  uint32_t baudrate;
+  /**
+   * @brief  RTS pin. 0 - disable.
+   */
+  gpio_t pin_rts;
+  /**
+   * @brief  RTS signals when transmit.
+   */
+  int pin_rts_enable;
+  /**
+   * @brief  ID device. 0 - master, otherwice slave.
+   */
+  uint8_t id;
+  /**
+   * @brief   Before slave begin to response; usec.
+   */
+  uint32_t timeout;
+  /**
+   * @brief   Template buffer for receive message or send. **Must never be changed
+   *          by the user.**
+   * @internal
+   */
+  uint8_t _buffer[MODBUS_RTU_PACKET_SIZE_MAX];
```

Comment:
I don't think that you need to add an underscore as prefix here. IMO all members of a device descriptor are internal and the user should treat its contents as opaque anyway, so no need to specifically mark them as internal. (E.g. if the user indeed would be allowed to modify other members, IMO there should be functions for accessing them added. This allows to change the data structure easier. You can use `static inline` functions where appropriate, so that this would not add any overhead.)